deploy/charts/cert-manager/templates/_helpers.tpl

```diff
@@ -172,3 +172,40 @@ https://github.com/helm/helm/issues/5358
 {{- define "cert-manager.namespace" -}}
     {{ .Values.namespace | default .Release.Namespace }}
 {{- end -}}
+
+{{- define "featureGatesController" -}}
+{{- $featureGatesController := list "AllAlpha" "AllBeta" "AdditionalCertificateOutputFormats" "ValidateCAA" "ExperimentalCertificateSigningRequestControllers" "ExperimentalGatewayAPISupport" "ServerSideApply" "LiteralCertificateSubject" "UseCertificateRequestBasicConstraints" "SecretsFilteredCaching" -}}
+{{- include "featureGates" (list $.Values.featureGates $featureGatesController) -}}
+{{- end -}}
+
+{{- define "featureGatesWebhook" -}}
+{{- $featureGatesWebhook := list "AllAlpha" "AllBeta" "AdditionalCertificateOutputFormats" "LiteralCertificateSubject" -}}
+{{- include "featureGates" (list $.Values.featureGates $featureGatesWebhook) -}}
+{{- end -}}
+
+{{- define "featureGatesCAInjector" -}}
+{{- $featureGatesCAInjector := list "AllAlpha" "AllBeta" "ServerSideApply" -}}
+{{- include "featureGates" (list $.Values.featureGates $featureGatesCAInjector) -}}
+{{- end -}}
+
+{{- define "featureGates" -}}
+{{- $featureGatesString := index . 0 -}}
+{{- $featureGates := index . 1 -}}
+{{- $featureGatesAll := list "AllAlpha" "AllBeta" "AdditionalCertificateOutputFormats" "ValidateCAA" "ExperimentalCertificateSigningRequestControllers" "ExperimentalGatewayAPISupport" "ServerSideApply" "LiteralCertificateSubject" "UseCertificateRequestBasicConstraints" "SecretsFilteredCaching" -}}
+{{- /**/ -}}
+{{- $flags := list -}}
+{{- range $gate := split "," (nospace $featureGatesString) -}}
+{{-     $gateName := (split "=" $gate)._0 -}}
+{{-     $gateValue := (split "=" $gate)._1 -}}
+{{-     if $gateName -}}
+{{-         if not (has $gateName $featureGatesAll) -}}
+{{-             fail (printf "The featureGate %s is unknown." $gateName) -}}
+{{-         end -}}
+{{-         if has $gateName $featureGates -}}
+{{-             $flags = append $flags (printf "%s=%s" $gateName $gateValue) -}}
+{{-         end -}}
+{{-     end -}}
+{{- end -}}
+{{- /**/ -}}
+{{- $flags | join "," -}}
+{{- end -}}
```

Comment:
Personally I would prefer to not add this kind of templating complexity to the helm chart if we can achieve the same with two/three simple fields in values file. We don't test the helm chart templating and this is going to be difficult to maintain and even just understand what it does.
What are you actually trying to achieve here?

Reply:
It fixes the issue of having to split the feature gates per component.
This logic is identical to what we had here: https://github.com/cert-manager/cert-manager/pull/6093/files#diff-97c9bba50f10f0b01d247ed02847c7a197184b76539cd056a738a87f74fe4003L238-L240

Reply:
But that's not the behavior we want, the feature gates should be separate for components, if I enable SSA for controller it should not also be enabled for cainjector for example. The test setup is our specific configuration where this makes sense, I don't think we should impose these restrictions for users in the helm chart

Reply:
We discussed this and agreed that we want to have multiple featureGates options in the values.yaml file (one for each binary).
Splitting the current field into multiple fields might break some people, but that is acceptable since these are Alpha/ Beta features anyway.

In a future PR we will also add a featureGates option for the CAInjector. In the release notes we will have a clear indication that this is a breaking change.

I think we should not backport this, so we don't break anyone. Also this has not caused much problems in 1.11.x, so 1.12.x will be able to deal with this issue too.

Reply:
I think _not_ backporting though means that more folks will start relying on the faulty behaviour